<launch>
    <arg name="debug_directory" value="$(env HELLO_FLEET_PATH)/debug/" />
    <!-- <include file="$(find sp_core)/launch/robot_nav_bringup.launch"></include> -->
    <include file="$(find sp_perception)/launch/rs.launch"></include>
    <!-- -->
    <!-- ARUCO MARKER DETECTOR -->
    <include file="$(find sp_perception)/launch/aruco.launch">
        <arg name="camera" default="/camera" />
        <arg name="fiducial_len" default="0.05" />
    </include>
    <!-- -->
    <include file="$(find icl_pupil_ros)/launch/pupil.launch">
        <arg name="draw_gaze" default="true"/>
    </include>

	<!-- WebSocket -->
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" />
	<!-- -->

    <!-- STRETCH DRIVER -->
    <param name="/stretch_driver/broadcast_odom_tf" type="bool" value="true" />
    <param name="/stretch_driver/mode" type="string" value="navigation" />
    <include file="$(find stretch_core)/launch/stretch_driver.launch" pass_all_args="true" />

    <!-- MAPPING -->
    <node name="funmap" pkg="stretch_funmap" type="funmap" output="screen">
        <param name="debug_directory" type="string" value="$(arg debug_directory)" />
    </node>
    <!-- -->

    
    <!-- LASER RANGE FINDER -->
    <include file="$(find sp_core)/launch/rplidar.launch" />
    <!--  -->
    <!-- LASER SCAN MATCHER FOR ODOMETRY -->
    <include file="$(find stretch_core)/launch/stretch_scan_matcher.launch" />
    <!-- -->

    <!-- <include file="$(find sp_navigation)/launch/nav_3d.launch"></include> -->
    

    <node pkg="sound_play" type="soundplay_node.py" name="soundplay_node"/>
    <node name="tts_node" pkg="sp_voice" type="tts_node.py" output="screen"></node>

    <node name="study_node" pkg="sp_core" type="study_node.py" output="screen"></node>

    <node pkg="robot_face" type="serve.py" name="robot_face_node" />

    <!-- <node pkg="laser" type="laser_node" name="laser_node" output="screen"/> -->

</launch>